### PR TITLE
#100 Events on home page filtered by favorite categories

### LINKED ISSUE #100: As a user I want to view the list of events on the Home page within my favorite categories
1. As a user I have the ability to choose my favorite categories while editing the profile.
2. Use this feature to filter the list of categories on my Home page.
3. In the case of an empty list, use some default criteria (most popular, not far in the future, etc).

### LINKED ISSUE #148: "Share event" button refer to incorrect url.

### PR DESCRIPTION
**Pull request by andrii-zaharovskyi**
Close #100 
Close #148

### CODE REVIEW

File: EventsExpress/ClientApp/src/actions/event-list.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
-
+import { data } from 'jquery';
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/actions/event-list.js]: Is this line required?

File: EventsExpress/ClientApp/src/components/event/event-cancel-modal.js [hunk 2]
```diff
@@ -53,9 +50,14 @@ class EventCancelModal extends Component {
                                 <input size="50" type='text' onChange={this.handleChange} />
                             </div>
                             {this.props.cancelationStatus.errorMessage &&
-                                <div style={{color: 'red'}}>
-                                {JSON.stringify(JSON.parse(this.props.cancelationStatus.errorMessage)["errors"]["Reason"][1])}
-                                </div>
+                                renderErrorMessage(this.props.cancelationStatus.errorMessage, 'Reason')
+                                // <div style={{ color: 'red' }}>
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/event/event-cancel-modal.js]: Uncomment these lines or remove them.

File: EventsExpress/ClientApp/src/containers/event-filter.js [hunk 3]
```diff
@@ -1,98 +1,97 @@
...
-                var categories = '';
-                for (var i = 0; i < filters.categories.length; i++) {
-                    categories += filters.categories[i].id + ',';
-                }
-                search_string += '&categories=' + categories;
-            }
-            if(filters.status==="all"){
-                search_string+='&All='+true;
+    onSubmit = (filters) => {
+        filters = JSON.parse(JSON.stringify(filters));
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/containers/event-filter.js]: Why is this required?

File: EventsExpress/ClientApp/src/services/EventsExpressService.js [hunk 4]
```diff
@@ -87,6 +90,7 @@ export default class EventsExpressService {
         localStorage.setItem('token', rest.jwtToken);
         return true;
     }
+    //#endregion Resource Interaction Interface
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/services/EventsExpressService.js:93]: Remove commented lines.

File: EventsExpress/EventsExpress.csproj [hunk 5]
```diff
@@ -25,7 +25,6 @@
     <PackageReference Include="MediatR" Version="7.0.0" />
     <PackageReference Include="MediatR.Extensions.Microsoft.DependencyInjection" Version="7.0.0" />
     <PackageReference Include="MicroElements.Swashbuckle.FluentValidation" Version="3.1.0" />
-    <PackageReference Include="Microsoft.AspNetCore.App" />
```
Comment (iuriikhrystiuk) [inline EventsExpress/EventsExpress.csproj:28]: Why was this removed?

**Review by iuriikhrystiuk — CHANGES_REQUESTED** [review]

File: EventsExpress/EventsExpress.csproj (hunk 5 above)
Comment (andrii-zaharovskyi) [reply EventsExpress/EventsExpress.csproj:28]: This is an obsolete package reference: [Migrate from ASP.NET Core 2.2](https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.1&tabs=visual-studio#update-the-project-file)

File: EventsExpress/ClientApp/src/components/event/event-cancel-modal.js (hunk 2 above)
Comment (andrii-zaharovskyi) [reply EventsExpress/ClientApp/src/components/event/event-cancel-modal.js]: The commented code has been removed.

File: EventsExpress/ClientApp/src/actions/event-list.js (hunk 1 above)
Comment (andrii-zaharovskyi) [reply EventsExpress/ClientApp/src/actions/event-list.js]: The outdated import has been removed.

File: EventsExpress/ClientApp/src/containers/event-filter.js (hunk 3 above)
Comment (andrii-zaharovskyi) [reply EventsExpress/ClientApp/src/containers/event-filter.js]: This is required to trim keys with undefined values. This code has been refactored.

**Review by iuriikhrystiuk — APPROVED** [review]
